controllers/storagecluster/cephcluster.go

```diff
@@ -1572,6 +1572,12 @@ func getCephClusterCephConfig(r *StorageClusterReconciler, sc *ocsv1.StorageClus
 		r.Log.Error(err, "could not get network config details", "CephCluster", klog.KRef(sc.Namespace, sc.Name))
 	}
 
+	if shouldUseHostNetworking(sc) {
+		cephConfig["client.rgw.*"] = map[string]string{
+			"rgw_frontends": fmt.Sprintf("beast endpoint=0.0.0.0:%d secure_port=%d", getRGWPort(sc), getRGWSecurePort(sc)),
```

Comment:
Have you verified that this actually changes RGW's behavior? This is given to RGW as a CLI flag which usually has a higher precedence than Ceph configs. I would expect this change to have no effect on RGW behavior.

Also, have youtried to set this on the CephObjectStore directly using `.gateway.rgwConfig` spec? Those are a better option since they are applied to only the specific object store and not all possible object stores.

Reply:
Config takes effect
```
$ ceph config dump | grep rgw_front
client.rgw.*                                           basic     rgw_frontends                          beast endpoint=0.0.0.0:50080 secure_port=50443  *
 ```

and RGW is listining on the correct port
```
 netstat -tlnp | grep rados
tcp        0      0 0.0.0.0:50080           0.0.0.0:*               LISTEN      187974/radosgw
tcp        0      0 0.0.0.0:50443           0.0.0.0:*               LISTEN      187974/radosgw
```

Reply:
`ceph config dump` doesn't guarantee that RGW is actually **using** the config. It only dumps the config database. You need to look at the RGW log or and/or use `ceph config show` to actually see the running config.

Reply:
> Applying `.gateway.rgwConfig` got removed

I'm not sure what you mean by that. Can you clarify?

If it's not being stored (or even if it is), I'd recommend overriding any frontend configs using `RgwCommandFlags`. And I'd recommend not overriding the `rgw_frontends` config unless it's needed. Overriding this makes the config unique among Rook object configs in our ODF product and could make debugging and support harder.

Reply:
I was looking for `.gateway.rgwConfig` in ceph config wich was missing but was working, but it turns out the issue was because of rook operator not using dnsPolicy `ClusterFirstWithHostNet`

Removed the frontend config change